vite.config.ts

```diff
@@ -0,0 +1,24 @@
+import { defineConfig } from 'vite'
+import react from '@vitejs/plugin-react'
+import { resolve } from 'path'
+import dts from 'vite-plugin-dts'
+
+
+
+// https://vitejs.dev/config/
+export default defineConfig({
+  plugins: [react(), dts()],
+  build: {
+    lib: {
+      "name": "@nilfoundation/ui-kit",
+      // Could also be a dictionary or array of multiple entry points
+      entry: resolve(__dirname, 'src/index.ts'),
+      formats: ['es', 'cjs'],
+    },
+    rollupOptions: {
+      // make sure to externalize deps that shouldn't be bundled
+      // into your library
+      external: ['react', 'react-dom', 'baseui'],
```

Comment:
No it don't need to be deifned here. It will be provided through providers so it will be accessible. I move it to dev library because we don't need it in production build. Library doesn't have any dependency now only peerDependencies